### ISSUE TITLE
Drop Db doesn't (can't) work if connected to the db you want to drop

### ISSUE THREAD

**Issue by vertiman** [db-migrate/pg#3]
The Drop DB command doesn't work if you're trying to drop the database which you're connected to.  You can get around this by creating another environment that connects to the template1 or postgres database, and then drop the one you want.  

I can't think of an intelligent way to bake this behaviour into the driver so we don't need to use a workaround, any ideas?  If you have an implementation idea I'm happy to take a stab at implementing it.

**Comment by wzrdtales** [comment]
Hum, ic. This is problematic indeed, quick thought, if it is not possible to delete the db you're connected to. I assume it is also not possible to delete a db as long as at least one 1 connection is open to it?

**Comment by vertiman** [comment]
Yeah, although I think you can issue commands to drop others connections.  Are you thinking establish a separate connection inside the dropDatabase on a different database?  Are all the connection parameters (login, password, host) available in that scope?

**Comment by wzrdtales** [comment]
Ok, I take a few minutes of my time now for a quick write up:

Since postgres is always only connected to one db per connection and it is not possible to switch (I already stumbled this when I implemented the scopes a while ago) dbs and now it opens that it is also not possible to delete dbs.

What I can think of is:
Open up a "maintenance" connection on the driver and connect to the default db to execute this kinds of queries. This connection gets either opened on demand and closed instantly after the action has taken place or is opened on the driver creation.
From the point that it is currently not implemented as a feature to switch the database when using scopes see https://github.com/db-migrate/pg/blob/master/index.js#L105-L121 and http://db-migrate.readthedocs.org/en/latest/Getting%20Started/the%20commands/#scope-configuration, it might be from interest to look at the possibility of creating new connections and disposing old ones, or replacing them, on the pg driver level.

**Comment by wzrdtales** [comment]
yes they are, they get passed as config object here https://github.com/db-migrate/pg/blob/master/index.js#L486

**Comment by vertiman** [comment]
Ok I'll fiddle with this over the next few days.  Need to work on something else first.  Thanks for the idea, it will probably work.  From a usage perspective, can you do a db:drop at the same time as any other command?  I don't think so, right?  So we don't have to worry about someone polluting the "main" connection, if I manage to programatically kill it?

**Comment by wzrdtales** [comment]
It would be possible, but I do not see a reason why one would execute a drop action in another instance of db-migrate while migrating something.
